hypertrace-gradle-java-convention-plugin/src/main/java/org/hypertrace/gradle/java/convention/JavaConventionPlugin.java

```diff
@@ -12,15 +12,23 @@
 public class JavaConventionPlugin implements Plugin<Project> {
 
   public void apply(Project target) {
+    // plugin meant to be applied at root level, it configures for all nested subprojects
     target
-        .getPluginManager()
-        .withPlugin(
-            "java",
-            unused -> {
-              createConvention(target);
-              configureCompatibility(target);
-              configureToolchain(target);
-              modifyTestJvmArgs(target);
+        .getSubprojects()
+        .forEach(
+            subproject -> {
+              subproject
+                  .getPluginManager()
+                  .withPlugin(
+                      "java",
+                      unused -> {
+                        createConvention(subproject);
+                        configureCompatibility(subproject);
+                        configureToolchain(subproject);
+                        modifyTestJvmArgs(subproject);
+                      });
+              // recursive call
+              apply(subproject);
```

Comment:
Not needed. all of these forms are already recursive (subprojects, allprojects return all descendants not just direct ones)